### DISCUSSION TITLE
Using #each with body template request helper

### DISCUSSION THREAD

**Discussion by robguyoncourt** [mockoon/mockoon#516] [Help] [answered]
I'm trying to iterate through part of the request body using "#each" to dynamically produce a response. For example with a request body like this:

`{"asDate":"2021-07-02","instruments":[{"identifiers":[{"id":"05VLW7","idType":"TICKER_US"}]},{"identifiers":[{"id":"AAPL","idType":"TICKER_US"}]}]}`

I'd like to iterate like this in the response template:
`{{#each (body 'instruments')}}
{{/each}}`

Unfortunately that only seems to produce a single iteration rather than the two I was expecting. What am I doing wrong?

**Comment by 255kb** [comment, marked as answer]
You are not doing anything wrong. The `body` helper stringifies objects and arrays it encounters. You are not the first one asking about this and I realize that it probably wasn't the best decision I made 😃 
What you need is to iterate with a `repeat` until with come with a better solution:

```
{{# repeat 2}}
  {{ body (concat 'instruments.' @index) }}
{{/ repeat}}
```

**Reply by robguyoncourt** [reply]
Thanks! Had come up with something similar using #repeat in combination with a #if to skip blanks. Good enough as a work-around.

**Reply by 255kb** [reply]
I created the issue anyway, I think it should definitely be improved mockoon/commons-server#18

**Reply by tomaslad** [reply]
@robguyoncourt please, could you share your current solution? I am facing the same problem :)

**Reply by robguyoncourt** [reply]
@tomaslad  pretty much as @255kb  suggested: I added the following which enables me to handle up to 50 repetitions (instruments in my request message), but prevents blank sections being added to the response. That's fine for my purposes:
```
{{#repeat 50}}
{{setVar 'ident' (body (concat 'instruments.' @index '.id'))}}
{{#if (concat @ident '')}}
```
